### ISSUE TITLE
Delete scalaz.effect.IO

### ISSUE THREAD

**Issue by Milyardo** [scalaz/scalaz#2038]
With Task gone, it's now time to look at IO. Instead of deleting scalaz.effect outright however, I thought it best to raise an issue to see if `MonadIO` or any typeclasses in effect are worth salvaging. If they are should should `scalaz-effect` as a module remain and should these typeclasses be moved to `core`?

**Comment by emilypi** [comment]
@Milyardo lets move the instances to `core`, and nuke `scalaz.effect`?

**Comment by tonymorris** [comment]
I agree, if it can be made to work. It was all the hack jobs that needed deletion, including Task.
